train.py

```diff
@@ -72,10 +85,65 @@ def loss_for_permutation(
             loss += best_loss
     return loss, best_mask_order
 
+
+def save_ckpt(ckpt_file: str, model: nn.Module,
+              optimizer: Optional[th.optim.Optimizer] = None):
+    ckpt_file = Path(ckpt_file)
+    _ensure_dir(ckpt_file.parent)
+    save_dict: Dict[str, Any] = {}
+    save_dict['model'] = model.state_dict()
+    if optimizer is not None:
+        save_dict['optimizer'] = optimizer.state_dict()
+    th.save(save_dict, str(ckpt_file))
+
+
+def load_ckpt(ckpt_file: str, model: nn.Module,
+              optimizer: Optional[th.optim.Optimizer] = None):
+    ckpt_file = Path(ckpt_file)
+    save_dict = th.load(str(ckpt_file))
+    model.load_state_dict(save_dict['model'])
+    if optimizer is not None:
+        optimizer.load_state_dict(save_dict['model'])
```

Comment:
I noticed that this should be `save_dict['optimizer']` instead.
When pausing and then resuming training, this sounds fine. However, I think the optimizer state shouldn't be loaded when resuming with different settings(i.e., entering the 2nd stage once the 1st stage of the 3-stage training is done). How should we handle this?

Reply:
(1) Oops, that's a bug...
(2) I think it should be controlled by an argument, like `model.load(..., load_optimizer=[True,False])`.

Reply:
> I think it should be controlled by an argument, like `model.load(..., load_optimizer=[True,False])`.

Sounds good. Do you want to make a PR which addresses my comments?

Reply:
Sure. Wait a sec...

Reply:
#11